### PR TITLE
testing

### PR DESCRIPTION
**Pull request by EntyR**

### CODE REVIEW

File: domain/build.gradle [hunk 1]
```diff
@@ -20,4 +20,6 @@ tasks {
 dependencies {
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.5.1")
     implementation("io.insert-koin:koin-core:3.1.2")
+
+    testImplementation 'junit:junit:4.13.2'
```
Comment (AlexeyKolosov) [inline domain/build.gradle:32]: lets use these:
```
    "org.junit.jupiter:junit-jupiter-api:$version"
    "org.junit.jupiter:junit-jupiter-engine:$version"
    "org.junit.jupiter:junit-jupiter-params:$version"
    "com.google.truth:truth:$version"  //convenient assert function
    "io.mockk:mockk:$version"    //mocking library
```

File: domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.harman.roomdbapp.domain.use_cases
+
+import org.junit.Test
```
Comment (AlexeyKolosov) [inline domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt]: for new dependencies,  you can use 
`import org.junit.jupiter.api.Test`

File: domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+internal class RandomNumberUseCaseTest {
+
+
```
Comment (AlexeyKolosov) [inline domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt:25]: private lateinit var repository: IRandomNumberRepository

    @BeforeEach     //import org.junit.jupiter.api.BeforeEach
    fun setUp() {
        repository= mockk(relaxed = true) // import io.mockk.mockk
    }

File: domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    @Test
+    fun addNumber() {
+
+    }
+
+    @Test
```
Comment (AlexeyKolosov) [inline domain/src/test/java/com/harman/roomdbapp/domain/use_cases/RandomNumberUseCaseTest.kt:44]:
```
@Test
fun `Verify numbers are returned from repository`() = runBlocking {
    every { repository.getNumbers() } returns flowOf(numbers)
    val useCase = RandomNumberUseCase(repository)
    assertThat(useCase.getNumbers().take(1).toList()).containsExactly(numbers).inOrder()
}
```

File: data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @Test(expected = Test.None::class)
+    fun verify_if_flow_can_be_received_from_database_without_exception() {
+        db.getDao().getAllNumbers()
+    }
+
+
+    @Test(expected = Test.None::class)
+    fun verify_if_number_can_be_inserted_into_database_without_exception() {
+
+        db.getDao().getAllNumbers()
```
Comment (AlexeyKolosov) [inline data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt]: This is a duplicate, remove if it is not needed

File: data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Test
+    fun verify_if_number_was_added_and_can_be_received_from_database() = runBlocking {
+
+        val number = RandomNumberEntity(0, 1)
+        db.getDao().addNumber(number)
+        val result = db.getDao().getAllNumbers().firstOrNull {
+            it.contains(number)
+        }
+        Truth.assertThat(result).isNotNull()
```
Comment (AlexeyKolosov) [inline data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt]:
```
    @Test
    fun verify_if_number_was_added_and_can_be_received_from_database() = runBlocking {
        val numbers = listOf(RandomNumberEntity(0, 1))
        numbers.forEach {
            db.getDao().addNumber(it)
        }
        val events = mutableListOf<RandomNumberEntity>()
        db.getDao().getAllNumbers().take(1).collect {
            events.addAll(it)
        }
        Truth.assertThat(events.size).isEqualTo(numbers.size)
        events.forEachIndexed{ index, entity ->
            Truth.assertThat(entity.number).isEqualTo(numbers[index].number)
        }
    }
```

File: data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+internal class EntityMappersTest {
+
+    @Test
+    fun `Verify conversion to Entity was right`() {
+
+        val expected = RandomNumberEntity(0, 1)
+        val initial = RandomNumber(1)
+
+        Truth.assertThat(initial.convertToEntity()).isEqualTo(expected)
```
Comment (AlexeyKolosov) [inline data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt]: please add `assertThat `for all fields of `initial.convertToEntity()` object, like this:
```
val actual = initial.convertToEntity()
assertThat(actual.id).isEqualTo(expected.id)
assertThat(actual.number).isEqualTo(expected.number)
```

File: data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    }
+
+    @Test
+    fun `Verify conversion to Number was right`() {
+        val initial = RandomNumberEntity(0, 1)
+        val expected = RandomNumber(1)
+
+        Truth.assertThat(initial.convertToRNumber()).isEqualTo(expected)
```
Comment (AlexeyKolosov) [inline data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt]: the same here 
` Truth.assertThat(initial.convertToRNumber().number).isEqualTo(expected.number)`

**Comment by AlexeyKolosov** [comment]
Also, I would suggest changing:
```
@Entity
data class RandomNumberEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,    // made this optional
    val number: Int
)
fun RandomNumber.convertToEntity(): RandomNumberEntity {
    return RandomNumberEntity(
        number = this.number   // and there is no need to set id=0
    )
}
```

File: data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt (hunk 7 above)
Comment (EntyR) [reply data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt]: done

File: data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt (hunk 8 above)
Comment (EntyR) [reply data/src/test/java/com/harman/roomdbapp/data/EntityMappersTest.kt]: done

File: data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt (hunk 6 above)
Comment (EntyR) [reply data/src/androidTest/java/com/harman/roomdbapp/data/repository/RandomNumberRepositoryTest.kt]: done

**Pull request merged**